sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java

```diff
@@ -170,11 +173,12 @@ public WriteResult expand(PCollection<KV<DestinationT, TableRow>> input) {
                     new DoFn<Void, String>() {
                       @ProcessElement
                       public void getTempFilePrefix(ProcessContext c) {
-                        c.output(
-                            resolveTempLocation(
-                                c.getPipelineOptions().getTempLocation(),
-                                "BigQueryWriteTemp",
-                                stepUuid));
+                        String tempLocation = resolveTempLocation(
+                            c.getPipelineOptions().getTempLocation(),
+                            "BigQueryWriteTemp",
+                            BigQueryHelpers.randomUUIDString());
+                        LOG.info("BigQuery temporary file location {}", tempLocation);
```

Comment:
"Writing BigQuery temporary files to {} before importing them"

(or something similar that describes what we're doing?)